File: typescript/express-example/src/routes/tts.ts
import express, { Request, Response } from 'express';
import { CreoleCentricAPI, TTSJobRequest } from '../lib/creolecentric.js';

const router = express.Router();

// Initialize API client
const apiKey = process.env.CREOLECENTRIC_API_KEY;
const apiUrl = process.env.CREOLECENTRIC_API_URL;

if (!apiKey) {
  throw new Error('CREOLECENTRIC_API_KEY environment variable is required');
}

const client = new CreoleCentricAPI({
  apiKey,
  baseUrl: apiUrl,
});

// Health check
router.get('/health', async (req: Request, res: Response) => {
  try {
    const health = await client.checkHealth();
    res.json(health);
  } catch (error) {
    throw error;
  }
});

// Get credit balance
router.get('/credits/balance', async (req: Request, res: Response) => {
  try {
    const balance = await client.getCreditBalance();
    res.json(balance);
  } catch (error) {
    throw error;
  }
});

// Get available voices
router.get('/tts/voices', async (req: Request, res: Response) => {
  try {
    const voices = await client.getVoices();
    res.json(voices);
  } catch (error) {
    throw error;
  }
});

// Get available models
router.get('/tts/models', async (req: Request, res: Response) => {
  try {
    const models = await client.getModels();
    res.json(models);
  } catch (error) {
    throw error;
  }
});

// Create TTS job
router.post('/tts/jobs', async (req: Request, res: Response) => {
  try {
    const jobRequest: TTSJobRequest = req.body;

    // Validate required fields
    if (!jobRequest.text || jobRequest.text.trim() === '') {
      return res.status(400).json({ error: 'Text is required' });
    }

    const job = await client.createTTSJob(jobRequest);
    res.json(job);
  } catch (error) {
    throw error;
  }
});

// Get job status
router.get('/tts/jobs/:jobId', async (req: Request, res: Response) => {
  try {
    const { jobId } = req.params;

    if (!jobId) {
      return res.status(400).json({ error: 'Job ID is required' });
    }

    const job = await client.getJobStatus(jobId);
    res.json(job);
  } catch (error) {
    throw error;
  }
});

// Get job details
router.get('/tts/jobs/:jobId/details', async (req: Request, res: Response) => {
  try {
    const { jobId } = req.params;

    if (!jobId) {
      return res.status(400).json({ error: 'Job ID is required' });
    }

    const job = await client.getJobDetails(jobId);
    res.json(job);
  } catch (error) {
    throw error;
  }
});

// List jobs
router.get('/tts/jobs', async (req: Request, res: Response) => {
  try {
    const limit = parseInt(req.query.limit as string) || 10;
    const offset = parseInt(req.query.offset as string) || 0;

    const jobs = await client.listJobs(limit, offset);
    res.json(jobs);
  } catch (error) {
    throw error;
  }
});

// Cancel job
router.post('/tts/jobs/:jobId/cancel', async (req: Request, res: Response) => {
  try {
    const { jobId } = req.params;

    if (!jobId) {
      return res.status(400).json({ error: 'Job ID is required' });
    }

    const result = await client.cancelJob(jobId);
    res.json(result);
  } catch (error) {
    throw error;
  }
});

export default router;

File: python/flask-example/routes/tts.py
from flask import Blueprint, request, jsonify
from services.creolecentric import CreoleCentricClient
import requests as http_requests

tts_bp = Blueprint('tts', __name__)
client = CreoleCentricClient()


@tts_bp.route('/tts/jobs', methods=['POST'])
def create_tts_job():
    """
    Create a new TTS job

    Accepts optional callback_url parameter for webhook notifications.
    Using webhooks is RECOMMENDED over polling for production applications.
    """
    try:
        data = request.get_json()

        if not data or 'text' not in data:
            return jsonify({'error': 'Text is required'}), 400

        # Get callback URL from request if provided
        callback_url = data.get('callback_url')

        job = client.create_tts_job(
            text=data['text'],
            voice_id=data.get('voice_id', 'i4mRPwKM2yHwXhbmkN514'),
            model_id=data.get('model_id', 'ccl_ht_v100'),
            callback_url=callback_url,
            speed=data.get('speed'),
            pitch=data.get('pitch')
        )

        return jsonify(job), 201

    except http_requests.HTTPError as e:
        return jsonify({
            'error': str(e),
            'details': e.response.text if e.response else None
        }), e.response.status_code if e.response else 500

    except Exception as e:
        return jsonify({'error': str(e)}), 500


@tts_bp.route('/tts/jobs/<job_id>', methods=['GET'])
def get_job_status(job_id):
    """Get job status"""
    try:
        status = client.get_job_status(job_id)
        return jsonify(status)

    except http_requests.HTTPError as e:
        return jsonify({
            'error': str(e),
            'details': e.response.text if e.response else None
        }), e.response.status_code if e.response else 500

    except Exception as e:
        return jsonify({'error': str(e)}), 500


@tts_bp.route('/tts/voices', methods=['GET'])
def get_voices():
    """Get available voices"""
    try:
        voices = client.get_voices()
        return jsonify(voices)

    except http_requests.HTTPError as e:
        return jsonify({'error': str(e)}), e.response.status_code if e.response else 500

    except Exception as e:
        return jsonify({'error': str(e)}), 500


@tts_bp.route('/tts/models', methods=['GET'])
def get_models():
    """Get available models"""
    try:
        models = client.get_models()
        return jsonify(models)

    except http_requests.HTTPError as e:
        return jsonify({'error': str(e)}), e.response.status_code if e.response else 500

    except Exception as e:
        return jsonify({'error': str(e)}), 500


@tts_bp.route('/credits/balance', methods=['GET'])
def get_credit_balance():
    """Get credit balance"""
    try:
        balance = client.get_credit_balance()
        return jsonify(balance)

    except http_requests.HTTPError as e:
        return jsonify({'error': str(e)}), e.response.status_code if e.response else 500

    except Exception as e:
        return jsonify({'error': str(e)}), 500
